app/models/reviewable.rb

```diff
@@ -66,14 +66,11 @@ def self.default_visible
   end
 
   def self.valid_type?(type)
-    return false if Reviewable.types.exclude?(type)
-    type.constantize <= Reviewable
-  rescue NameError
-    false
+    type.to_s.safe_constantize.in?(types)
```

Comment:
TIL about `safe_constantize` and `in?` (which works even when called on `nil` 🤯)